mongo/integration/unified/collection_operation_execution.go

```diff
@@ -902,3 +854,85 @@ func buildUpdateResultDocument(res *mongo.UpdateResult) (bsoncore.Document, erro
 	}
 	return builder.Build(), nil
 }
+
+type cursorResult struct {
+	cursor *mongo.Cursor
+	err    error
+}
+
+func createFindCursor(ctx context.Context, operation *operation) (*cursorResult, error) {
```

Comment:
The code here is copied/pased from the existing `executeFind` method. The only difference is that it returns `*cursorResult` rather than `*operationResult` at the end because it's used as a helper function for both the `find` and `createFindCursor` operations.